### PR TITLE
Client/employee page search pagination

### PR DESCRIPTION
**Pull request by mglittleton**

### CODE REVIEW

File: client/src/Components/Dashboard/Dashboard.js [hunk 1]
```diff
@@ -78,7 +79,7 @@ class Dashboard extends Component {
                 />
                 <Route
                   path="/dashboard/employees"
-                  render={props => <EmployeeList {...props} />}
+                  render={() => <Employees />}
```
Comment (LBWright) [inline client/src/Components/Dashboard/Dashboard.js:91]: Be sure to spread props over the render so we have access to them in the page.

File: server/src/resources/employee/employee.model.js [hunk 2]
```diff
@@ -1,46 +1,46 @@
-import mongoose from 'mongoose';
+// import mongoose from 'mongoose';
```
Comment (LBWright) [inline server/src/resources/employee/employee.model.js]: Why not delete?

File: server/src/resources/employee/employee.router.js [hunk 3]
```diff
@@ -3,9 +3,11 @@ import { controllers } from './employee.controllers';
 
 const router = Router();
 
+
+
 router
   .route('/')
-  .get(controllers.getMany)
+  .get(controllers.getEmployees)
```
Comment (LBWright) [inline server/src/resources/employee/employee.router.js:8]: you still want this to be named `getMany`. Just overwrite the the class in the employees controller.

**Review by LBWright — CHANGES_REQUESTED** [review]

File: client/src/Components/Employees/EmployeeList.js [hunk 4]
```diff
@@ -9,59 +9,34 @@ class EmployeeList extends Component {
...
-    this.setState({ searchText: e.target.value });
-  };
-
   cardHandleClick = id => {
     this.props.history.push(`/dashboard/employees/${id}`);
   };
 
   render() {
+    const employees = this.props.employees;
+    const { pageSize } = this.props;
```
Comment (LBWright) [inline client/src/Components/Employees/EmployeeList.js]: Can you destructure both of these off of props?

File: client/src/Components/Employees/Employees.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    };
+  }
+
+  componentDidMount() {
+    const { filter, page, pageSize, sort } = this.state;
+    this.props.getEmployees({ page, pageSize, sort, filter });
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({
```
Comment (LBWright) [inline client/src/Components/Employees/Employees.js]: No reason to set state off of props

File: client/src/Components/Employees/Employees.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+  }
+
+  handleSearchChange = value => {
+    const { page, pageSize, sort, filter } = this.state;
+    const search = value || "";
+    this.setState({ search });
+    this.props.searchEmployees({ page, pageSize, sort, search, filter });
+  };
+
+  handleTabChange = (e, data) => {
```
Comment (LBWright) [inline client/src/Components/Employees/Employees.js]: you're using data here, but where are you passing it in?

File: client/src/Components/Employees/Employees.js [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+  };
+
+  render() {
+    const { tabs, page, pageSize } = this.state;
+    const { employees } = this.props
+
+    return (
+      <FlexColumn>
+        <Header as="h1">Employees</Header>
+        {this.props.loading ? (
```
Comment (LBWright) [inline client/src/Components/Employees/Employees.js]: Can you destructure loading off props with employees on line 54?

File: client/src/Components/Employees/Employees.js [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+              }
+            ]}
+          />
+        )}
+      </FlexColumn>
+    );
+  }
+}
+
+const mapStateToProps = state => {
```
Comment (LBWright) [inline client/src/Components/Employees/Employees.js]: You already set this inside of the container component

File: client/src/Components/Employees/Employees.js [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+  return {
+    employees: state.employees.employees,
+    loading: state.employees.loading,
+    error: state.employees.error
+  };
+};
+
+export default connect(
```
Comment (LBWright) [inline client/src/Components/Employees/Employees.js]: See above

File: client/src/Components/Employees/actionCreator.js [hunk 10]
```diff
@@ -2,22 +2,42 @@ import axios from "axios";
...
+
+  return axios
+    .get(
+      `${config.apiUrl}/api/employees?filter=${JSON.stringify(filter) ||
+        ""}&sort=${sort}&limit=${pageSize}&skip=${(page - 1) * pageSize}`
+    )
+    .then(({ data }) => {
+      dispatch({
+        type: actions.EMPLOYEE_SUCCESS,
+        employees: data.data
```
Comment (LBWright) [inline client/src/Components/Employees/actionCreator.js:18]: can you dispatch this as payload?
```
{
   type: actions.EMPLOYEE_SUCCESS,
   payload: data.data
}
```

File: client/src/Components/Employees/actionCreator.js [hunk 11]
```diff
@@ -2,22 +2,42 @@ import axios from "axios";
...
       });
-  };
+    })
+    .catch(err => {
+      dispatch({ type: actions.EMPLOYEE_FAILURE, error: err });
```
Comment (LBWright) [inline client/src/Components/Employees/actionCreator.js:22]: Same as above. Could you dispatch an action with a type and payload property?

File: client/src/Components/Employees/actionCreator.js [hunk 12]
```diff
@@ -2,22 +2,42 @@ import axios from "axios";
...
+  dispatch({ type: actions.EMPLOYEE_STARTED });
+
+  return axios
+    .get(
+      `${config.apiUrl}/api/employees/search?search=${search ||
+        ""}&filter=${JSON.stringify(filter) ||
+        ""}&sort=${sort}&limit=${pageSize}&skip=${(page - 1) * pageSize}`
+    )
+    .then(({ data }) => {
+      dispatch({ type: actions.EMPLOYEE_SUCCESS, employees: data.data });
```
Comment (LBWright) [inline client/src/Components/Employees/actionCreator.js:38]: See above

File: client/src/Components/Employees/actionCreator.js [hunk 13]
```diff
@@ -2,22 +2,42 @@ import axios from "axios";
...
+    })
+    .catch(err => {
+      dispatch({ type: actions.EMPLOYEE_FAILURE, error: err });
```
Comment (LBWright) [inline client/src/Components/Employees/actionCreator.js:41]: See above

File: client/src/Components/Employees/container.js [hunk 14]
```diff
@@ -1,16 +1,16 @@
-import { connect } from "react-redux";
-import { getEmployees } from "./actionCreator";
-import EmployeeList from "./EmployeeList";
+// import { connect } from "react-redux";
```
Comment (LBWright) [inline client/src/Components/Employees/container.js]: Please use the container.

File: client/src/Components/Employees/reducers.js [hunk 15]
```diff
@@ -17,7 +17,6 @@ const employeesReducer = (state = initialState, action) => {
       return {
         ...state,
         loading: false,
-        error: false,
         employees: action.employees
```
Comment (LBWright) [inline client/src/Components/Employees/reducers.js:20]: I mentioned changing this over to payload above. Would need to change it here too.
Payload is the Redux naming standard.

**Review by LBWright — CHANGES_REQUESTED** [review]

File: server/src/resources/employee/employee.router.js [hunk 16]
```diff
@@ -8,7 +8,11 @@ const router = Router();
 router
   .route('/')
   .get(controllers.getEmployees)
-  .post(controllers.createOne);
+  .post(controllers.createOne)
+
+router
+  .route('/num')
```
Comment (LBWright) [inline server/src/resources/employee/employee.router.js]: Interesting. What is this for?

Comment (mglittleton) [reply server/src/resources/employee/employee.router.js]: Nothing now. I found a different way to do the thing I want to do.

File: client/src/Components/Dashboard/Dashboard.js (hunk 1 above)
Comment (mglittleton) [reply client/src/Components/Dashboard/Dashboard.js:91]: I'm no longer going to be using props like this.

File: client/src/Components/Employees/actionCreator.js (hunk 11 above)
Comment (mglittleton) [reply client/src/Components/Employees/actionCreator.js:22]: I changed this on a later PR

File: client/src/Components/Employees/actionCreator.js (hunk 10 above)
Comment (mglittleton) [reply client/src/Components/Employees/actionCreator.js:18]: This was changed on a later PR.

File: client/src/Components/Employees/actionCreator.js (hunk 12 above)
Comment (mglittleton) [reply client/src/Components/Employees/actionCreator.js:38]: Changed later

File: client/src/Components/Employees/actionCreator.js (hunk 13 above)
Comment (mglittleton) [reply client/src/Components/Employees/actionCreator.js:41]: Changed later

File: client/src/Components/Employees/reducers.js (hunk 15 above)
Comment (mglittleton) [reply client/src/Components/Employees/reducers.js:20]: Changed later

**Review by juliejonak — APPROVED** [review]

**Review by LBWright — APPROVED** [review]
There are a few things that can be cleaned up, and doing logic on the employee reducer (causing a side effect) is an anti pattern, but it works for now.

**Pull request merged**